src/G2/Language/Naming.hs

```diff
@@ -433,15 +436,21 @@ instance Named Alt where
 
 instance Named DataCon where
     {-# INLINE names #-}
-    names (DataCon n t) = n S.<| names t
+    names (DataCon n t u e) = 
+        let 
+            all_tyvars = names u S.>< names e
+            t_ns = names t 
+        in 
+        n S.<| t_ns S.>< all_tyvars  
```

Comment:
Do we really need a let for these bindings?  Nitpicky, but sometimes just writing things inline is clearer, especially if lots of similar operations (just calling names on a bunch of stuff)